public/docs/_examples/cb-validation/ts/app/hero-form-model-binding.component.html

```diff
@@ -0,0 +1,75 @@
+ <!-- #docplaster -->
+ <!-- #docregion -->
+<div class="container">
+  <div [hidden]="submitted">
+    <h1>Hero Form (Model-Driven with Data Binding)</h1>
+    <form [ngFormModel]="heroForm" *ngIf="active" 
+          (ngSubmit)="onSubmit()">
+      <div class="form-group">
+        <!-- #docregion name-with-error-msg -->
+        <label for="name">Name</label>
+        <input type="text" class="form-control" 
+               [(ngModel)]="model.name"
+               ngControl="name"
+               [ngClass]="{'required': isRequired('name')}">
```

Comment:
Right - I understand the intent. My point is that these attributes are more than just hooks for initializing angular's validation - they also have a purpose purely within HTML. `maxlength` for instance will typically cause the browser to limit entry to the specified number of characters. `required` will be communicated via AT. If you were doing regex validation, the `pattern` attribute might determine what keyboard is used for entry on mobile devices. So if the model-driven technique doesn't render these attributes, the experience could be quite different than the template-driven technique, even if the validation itself is the same.